sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -614,27 +643,17 @@ int64_t CurlSession::OnRead(Context const& context, uint8_t* buffer, int64_t cou
   // check if all chunked is all read already
   if (this->m_isChunkedResponseType && this->m_chunkSize == this->m_sessionTotalRead)
   {
-    // Need to read CRLF after all chunk was read
-    for (int8_t i = 0; i < 2; i++)
-    {
-      if (this->m_bodyStartInBuffer > 0 && this->m_bodyStartInBuffer < this->m_innerBufferSize)
-      {
-        this->m_bodyStartInBuffer += 1;
-      }
-      else
-      { // end of buffer, pull data from wire
-        this->m_innerBufferSize = m_connection->ReadFromSocket(
-            context, this->m_readBuffer, Details::c_DefaultLibcurlReaderSize);
-        this->m_bodyStartInBuffer = 1; // jump first char (could be \r or \n)
-      }
-    }
+    ReadCRLF(context);
     // Reset session read counter for next chunk
     this->m_sessionTotalRead = 0;
     // get the size of next chunk
     ParseChunkSize(context);
 
     if (this->IsEOF())
-    { // after parsing next chunk, check if it is zero
+    {
+      // Final CRLF after end of chunk
+      ReadCRLF(context);
```

Comment:
How come we are reading CRLF twice, once above, and then again here?

Reply:
The first one on the top is the CRLF that is expected after the chunk data.
Then we read the next chunk size by calling `ParseChunkSize` and if the size is 0, we expect to read CRLF one more time to complete the transfer.

Reply:
@vhvb1989 worth putting a summary comment in the code to explain the logic and why it must be done.